compiler/rustc_span/src/hygiene.rs

```diff
@@ -46,6 +46,12 @@ use crate::symbol::{Symbol, kw, sym};
 use crate::{DUMMY_SP, HashStableContext, Span, SpanDecoder, SpanEncoder, with_session_globals};
 
 /// A `SyntaxContext` represents a chain of pairs `(ExpnId, Transparency)` named "marks".
+///
+/// This can generally be thought of as the macro call stack for a given location in code.
+/// Although, when a macro emits code from input parameters, that emitted code retains its
+/// call-site SyntaxContext.
+///
+/// See [<https://rustc-dev-guide.rust-lang.org/macro-expansion.html>] for more explanation.
```

Comment:
```suggestion
/// See <https://rustc-dev-guide.rust-lang.org/macro-expansion.html> for more explanation.
```

https://github.com/rust-lang/rust/issues/146162